### PR TITLE
Insights/shareable-link-modal

### PR DESCRIPTION
**Pull request by unclejustin**
Closes https://github.com/sourcegraph/sourcegraph/issues/34766

## Test plan

- Make sure that you can see the "share insight modal UI" by clicking on the context menu item - get a sharable link
- Make sure that the content of this UI is different based on a subset of dashboards in which this insight is included (if it's included in any global dashboard, you see that anyone can access this dashboard. If it's included in any org dashboard, you see a list of organizations that users should be part of to be able to access this insight by share link, and if it's is included in personal dashboards or in no dashboards, you should see that this insight is private and only you can access it by sharable link) See the original issue for more details about this states https://github.com/sourcegraph/sourcegraph/issues/34766
- Make sure that you copy that link and see the standalone insight page by this link

## App preview:

- [Web](https://sg-web-insights-shareable-link-modal.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gafooqqxcd.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by unclejustin** [comment]
@vovakulikov I got this one started for you.

Needs the tooltip on copy and to render the organization names instead of ids. I've asked the BE team if they could make the names available in the `insightViews` query.

File: client/web/src/enterprise/insights/components/modals/ShareLinkModal.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+const ShareLinkModalContent: React.FunctionComponent<Pick<ShareLinkModalProps, 'insight' | 'dashboard'>> = ({
+    insight,
+    dashboard,
+}) => {
+    if (!dashboard) {
+        return <GlobalContent />
+    }
+
+    const currentDashboard = insight.dashboards?.nodes.find(insightDashboard => insightDashboard.id === dashboard.id)
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/components/modals/ShareLinkModal.tsx:61]: @unclejustin, I don't this is what we want to have here. My original understanding was that in this modal, we should render a list of dashboards in which this insight is included.

**Comment by vovakulikov** [comment]
@unclejustin, suddenly I can't add you to the reviewers of this PR because you're an author of this PR.

**Comment by unclejustin** [comment]
@vovakulikov makes sense. Your changes look good to me though! 🙇 

I believe you can still review this one.

**Review by vovakulikov — APPROVED** [review]

**Comment by mrnugget** [comment]
@vovakulikov this fails on `main`: https://buildkite.com/sourcegraph/sourcegraph/builds/148141

screenshot_2022-05-18_09 54 37@2x

**Pull request merged**